shell.nix

```diff
@@ -0,0 +1,23 @@
+with import <nixpkgs> { };
+
+let
+  moz_overlay = import (builtins.fetchTarball
+    "https://github.com/mozilla/nixpkgs-mozilla/archive/master.tar.gz");
+  nixpkgs = import <nixpkgs> { overlays = [ moz_overlay ]; };
+in mkShell rec {
+  name = "eww";
+
+  buildInputs = [
+    glib
+    cairo
+    atk
+    gdk-pixbuf
+    pango
+    gtk3
+    (nixpkgs.rustChannelOf {
+      date = "2021-03-16";
+      channel = "nightly";
+    }).rust
+  ];
+  nativeBuildInputs = [ pkgconfig wrapGAppsHook ];
```

Comment:
```
RUST_SRC_PATH = "${nixpkgs.rust.packages.stable.rustPlatform.rustLibSrc}";
```
this line is needed so that rust-analyzer will work inside shell.nix